fastlane_bot/bot.py

```diff
@@ -357,9 +357,24 @@ def _convert_trade_instructions(
             for ti in errorless_trade_instructions_dic
             if ti is not None
         )
+        result = self._add_strategy_id_to_trade_instructions_dic(result)
         result = [TradeInstruction(**ti) for ti in result]
         return result
 
+    def _add_strategy_id_to_trade_instructions_dic(
+        self, trade_instructions_dic: Generator
+    ) -> list:
+        lst = []
+        for ti in trade_instructions_dic:
+            cid = ti["cid"]
+            if '-' in cid:
+                cid = cid.split('-')[0]
+            ti["strategy_id"] = self.db.get_pool(
+                cid=cid
+            ).strategy_id
+            lst.append(ti)
+        return lst
```

Comment:
TBH, I find the cluster of these three functions a little bit hard to follow:
1. `_drop_error`
2. `_convert_trade_instructions`
3. `_add_strategy_id_to_trade_instructions_dic`

And since two of them (1 and 3) are used only in a single place in the code, I suggest merging them into a single function whose code would be IMO easier to follow.

Something like this:
```
def _convert_trade_instructions(self, trade_instructions_dicts):
    errorless_trade_instructions_dicts = [
        {k: v for k, v in trade_instructions_dicts[i].items() if k != "error"}
        for i in range(len(trade_instructions_dicts))
    ]

    tis = [
        {
            **ti,
            "raw_txs": "[]",
            "pair_sorting": "",
            "ConfigObj": self.ConfigObj,
            "db": self.db,
            "strategy_id": self.db.get_pool(cid=ti["cid"].split('-')[0]).strategy_id
        }
        for ti in errorless_trade_instructions_dicts
        if ti is not None
    ]

    return [TradeInstruction(**ti) for ti in tis]
```

Reply:
this has been corrected